packages/autorest.typescript/CONTRiBUTING.md

```diff
@@ -0,0 +1,348 @@
+- `generate-version-tolerance:tests`: generates `updated` spec.ts based on hand written tests against the initial SDK
+- `test-version-tolerance`: executes all .spec.ts tests under `test/version-tolerance/`
+
+In CI we'll run `npm run validate-version-tolerance` which calls all the other npm scripts. The other individual npm scripts are useful for development.
+
+For more details on the implementation see [#1268](https://github.com/Azure/autorest.typescript/pull/1268)
+
+### How to add an integration test case
+
+Whenever you work on adding a feature/fixing a bug, this would probably be your first step. You create a test case and then run it through the generator, see the result, modify the generator, run it again and so on, until you get the desired output.
+
+1. Create your test input. Below are some examples
+   1. A simple [Swagger/OpenAPI file](./test/integration/swaggers/license-header.json))
+   2. A standalone [configuration markup file](./test/integration/swaggers/bodyComplex.md) (pointing to Swagger file somewhere else)
+   3. A combination of configuration markup file and Swagger/OpenAPI file.
+
+Let us say your test input will be called `testUserCase.json`.
+
+2. Now add the test input to the integration test suite to the file [`test-swagger-gen.ts`](./test/commands/test-swagger-gen.ts). In the file, add the following to the array `testSwaggers`:
+
+```
+testUserCase: {
+    swaggerOrConfig: "test/integration/swaggers/testUserCase.json",
+    clientName: "TestUserCaseClient",
+    packageName: "test-user-case",
+    licenseHeader: true,
+    useCoreV2: true,
+    allowInsecureConnection: true,
+    addCredentials: false
+},
+```
+
+3. Now, You can generate the SDK only for your test case with the following command: (Initially, during your development, you do not want to run all the cases during every step of your development. But, once your code changes are complete for your case, then you need to run the entire suite to ensure that your changes did not cause any unwanted changes.)
+
+```
+npm run generate-swaggers -- -b -i testUserCase
+```
+
+4. Once you are satisfied with the generated code, you can add a spec file such as `testUserCase.spec.ts` file [here](./test/integration). You can find several examples in the same place.
+
+### How to debug an integration test case
+
+#### `generate-swaggers` step
+
+If you would like to debug the `generate-swaggers` step for our test input, use the following command:
+
+```
+npm run generate-swaggers -- -b -i testUserCase --debug
+```
+
+Now, the code will wait for the debugger to be attached. Open the repository in VS Code -> Select `Run and Debug` section -> Click `Attach`.
+
+#### Spec file
+
+If you would like to debug the `testUserCase.spec.ts` file (after the SDK is generated), Open the repository in VS Code -> Open the `testUserCase.spec.ts` file -> Select `Run and Debug` section -> Click `IntegrationTests - Current File`.
```

Comment:
I think the name of `IntegrationTests - Current File` doesn't work anymore.
Could you update the launch.json to adapt this content?